circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit.cpp

```diff
@@ -65,6 +67,17 @@ void initialise_end_values(PrivateInputs<NT> const& private_inputs, KernelCircui
     end.new_commitments = start.new_commitments;
     end.new_nullifiers = start.new_nullifiers;
 
+    if (first_iteration) {
+        // Since it's the first iteration, we need to push the the tx hash nullifier into the `new_nullifiers` array
+
+        // If the nullifiers array is not empty a change was made and we need to rework this
+        if (!is_array_empty(end.new_nullifiers)) {
+            throw_or_abort("new_nullifiers array must be empty");
```

Comment:
You'll probably want to use `composer.do_assert()` as we do elsewhere in this file. Probably need to add a new error code.